spring-data-jpa/src/main/java/org/springframework/data/jpa/repository/support/JpaMetamodelEntityInformation.java

```diff
@@ -51,26 +50,28 @@
  * @author Christoph Strobl
  * @author Mark Paluch
  * @author Jens Schauder
+ * @author Greg Turnquist
  */
 public class JpaMetamodelEntityInformation<T, ID> extends JpaEntityInformationSupport<T, ID> {
 
 	private final IdMetadata<T> idMetadata;
 	private final Optional<SingularAttribute<? super T, ?>> versionAttribute;
 	private final Metamodel metamodel;
 	private final @Nullable String entityName;
+	private final PersistenceUnitUtil persistenceUnitUtil;
 
 	/**
 	 * Creates a new {@link JpaMetamodelEntityInformation} for the given domain class and {@link Metamodel}.
-	 *
+	 * 
 	 * @param domainClass must not be {@literal null}.
-	 * @param metamodel must not be {@literal null}.
+	 * @param entityManager#metamodel must not be {@literal null}.
 	 */
-	public JpaMetamodelEntityInformation(Class<T> domainClass, Metamodel metamodel) {
+	public JpaMetamodelEntityInformation(Class<T> domainClass, EntityManager entityManager) {
```

Comment:
Does it make sense to rather change the constructor to just add `PersistenceUnitUtil` as an additional parameter to avoid the need to depend on an `EntityManager` and move the preparation of the parameters to `JpaEntityInformationSupport `? This should also ease testing as we don't have to mock the intermediate mocking steps.